### ISSUE TITLE
cannot pass meta via ReferenceInput/useReferenceInputController

### ISSUE THREAD

**Issue by stesie** [marmelab/react-admin#8191]
**What you were expecting:**

Receiving `meta` in data provider when using `ReferenceInput` and passing in `queryOptions`.

**What happened instead:**

`meta` is undefined.

`ReferenceInput` passes all props towards `useReferenceInputController`.
Then `useReferenceInputController` destructures its props, among others into `queryOptions`.

Those `queryOptions` are then *all* passed to `useGetList` in https://github.com/marmelab/react-admin/blob/master/packages/ra-core/src/controller/input/useReferenceInputController.ts#L101
... however `meta` would have to go to the second parameter object.  Not the third.

Contrary `useListController` is separating those here https://github.com/marmelab/react-admin/blob/master/packages/ra-core/src/controller/list/useListController.ts#L48 (and passing separately in turn)

**Steps to reproduce:**

**Related code:**

```
                    <ReferenceInput source="pricelistId" reference="pricelists" queryOptions={{ meta }}>
                        <SelectInput fullWidth optionText="name" disabled />
                    </ReferenceInput>
```

**Other information:**

**Environment**

* React-admin version: 4.2.7
* Last version that did not exhibit the issue (if applicable):
* React version: 18.2.0
* Browser: Chrome 105
* Stack trace (in case of a JS error):

**Comment by WiXSL** [comment]
Thanks for the report!
Indeed, `ReferenceInput` has yet to have this feature.

**Comment by fzaninotto** [comment]
Fixed by #8132
